GPU/Common/VertexShaderGenerator.cpp

```diff
@@ -534,6 +534,11 @@ bool GenerateVertexShader(const VShaderID &id, char *buffer, const ShaderLanguag
 			WRITE(p, "uniform lowp float u_rotation;\n");
 		}
 
+#ifdef OPENXR
+		WRITE(p, "uniform lowp float u_scaleX;\n");
+		WRITE(p, "uniform lowp float u_scaleY;\n");
+#endif
```

Comment:
I would really like as many of these "ifdef OPENXR" to instead be something dynamic like g_Config.VREnabled or similar. You could have that as a config option that's just forced to true or false depending on the compile time flag for now. Future VR support for Windows will be in the main binary and will thus be toggle-able, so this needs to be dynamic. Also, maybe vec2 u_scaleXY?

Reply:
Fair enough. Do the shaders recompile if I change config during the gameplay?

Reply:
They'll be compiled anew when you start the next game. You're not generally gonna toggle between VR and non-VR at other times, right?

Reply:
Actually, it is possible to go into pause menu and disable VR (then it runs in a "cinema" mode).

However I can use DEVICE_TYPE_VR in this case.

Reply:
There's a shader cache, so it would need to be part of the shader ID to be sure it gets recompiled.

-[Unknown]

Reply:
Weeeeelll... Not really. The shader cache just stores the IDs so it can compile the shaders on startup, and does not yet keep generated shaders around between game executions. So it's actually not needed in this case, though highly recommended.